avocado/core/future/settings.py

```diff
@@ -0,0 +1,493 @@
+            is False.
+
+        action :
+            The basic type of action to be taken when this argument is
+            encountered at the command line. For more information visit the
+            argparser documentation.
+
+        .. note:: Most of the arguments here (like parser, positional_arg,
+                  short_arg, long_arg, choices, nargs, metavar, required and
+                  action) are only necessary if you would like to add a
+                  command-line option.
+        """
+        namespace = "{}.{}".format(section, key)
+        # Check if namespace is already registered
+        if namespace in self._namespaces:
+            msg = "Key {} already registered in section {}".format(key,
+                                                                   section)
+            raise DuplicatedNamespace(msg)
+
+        option = ConfigOption(namespace, help_msg, key_type, default,
+                              parser, short_arg, long_arg, positional_arg,
+                              choices, nargs, metavar, required, action)
+
+        # Register the option to a dynamic in-memory namespaces
+        self._namespaces[namespace] = option
+
+        if not parser:
+            return  # Nothing else to do here
+
+        if positional_arg:
+            parser.add_argument(namespace, **option.arg_parse_args)
+        else:
+            parser.add_argument(*option.name_or_tags, **option.arg_parse_args)
+
+    def update_option(self, namespace, value, convert=False):
+        """Convenient method to change the option's value.
+
+        This will update the value on Avocado internals and if necessary the
+        type convertion will be realized.
+
+        For instance, if an option was registered as bool and you call:
+
+            settings.register_option(namespace='foo.bar', value='true',
+                                     convert=True)
+
+        This will be stored as True, because Avocado will get the 'key_type'
+        registered and apply here for the conversion.
+
+        This method is useful when getting values from config files where
+        everything is stored as string and a conversion is needed.
+
+        Arguments
+        ---------
+        namespace : str
+            Your section plus your key, separeted by dots. The last
+            part of the namespace is your key. i.e: run.outputs.json.enabled
+            (section is `run.outputs.json` and key is `enabled`)
+
+        value : any type
+            This is the new value to update.
+
+        convert : bool
+            If Avocado should try to convert the value and store it as the
+            'key_type' specified during the register. Default is False.
+        """
+        if namespace not in self._namespaces:
+            return
+
+        option = self._namespaces[namespace]
+        option.set_value(value, convert)
+
+
+settings = Settings()  # pylint: disable-msg=invalid-name
```

Comment:
Consider adding a unit test that will, among testing it, show some bare use cases.  Example (feel free to reuse it): https://github.com/clebergnu/avocado/commit/d621d298158f0967b7140e0c61896dc2aec7bbd3

Reply:
Thanks for this. I really appreciate! 

I have changed a few things, but I will use it. Resolving this.